contracts/gateway/StakerProxy.sol

```diff
@@ -0,0 +1,202 @@
+            "This function can only be called by the composer."
+        );
+
+        _;
+    }
+
+    /** Requires that msg.sender is the owner. */
+    modifier onlyOwner() {
+        require(
+            msg.sender == owner,
+            "This function can only be called by the owner."
+        );
+
+        _;
+    }
+
+    /** Mutex to prevent reentrancy attack. */
+    modifier mutexed() {
+        assert(!stakeMutex);
+        stakeMutex = true;
+
+        _;
+
+        stakeMutex = false;
+    }
+
+
+    /* Special Functions */
+
+    /**
+     * @notice Must be constructed by a contract that implements the
+     *         `ComposerInterface`.
+     *
+     * @param _owner The owner that this proxy is deployed for.
+     */
+    constructor(address payable _owner) public {
+        composer = ComposerInterface(msg.sender);
+        owner = _owner;
+    }
+
+
+    /* External Functions */
+
+    /**
+     * @notice Initiates the stake process. In order to stake, the stake and
+     *         bounty amounts must first be transferred to the StakerProxy.
+     *         Staked amount is then transferred to the Gateway contract.
+     *         Bounty amount is also transferred to the Gateway contract.
+     *
+     * @param _amount Stake amount that will be transferred.
+     * @param _beneficiary The address in the auxiliary chain where the utility
+     *                     tokens will be minted.
+     * @param _gasPrice Gas price that staker is ready to pay to get the stake
+     *                  and mint process done.
+     * @param _gasLimit Gas limit that staker is ready to pay.
+     * @param _nonce The nonce to verify it is as expected.
+     * @param _hashLock Hash lock provided by the facilitator.
+     * @param _gateway On which gateway to stake.
+     *
+     * @return messageHash_ Message hash is unique for each request.
+     */
+    function stake(
+        uint256 _amount,
+        address _beneficiary,
+        uint256 _gasPrice,
+        uint256 _gasLimit,
+        uint256 _nonce,
+        bytes32 _hashLock,
+        EIP20GatewayInterface _gateway
+    )
+        external
+        onlyComposer
+        mutexed
+        returns (bytes32 messageHash_)
+    {
+        uint256 nonce = _gateway.getNonce(address(this));
+        require(
+            nonce == _nonce,
+            "Nonce must match nonce expected by gateway."
+        );
+
+        approveTransfers(_gateway, _amount);
+
+        messageHash_ = _gateway.stake(
+            _amount,
+            _beneficiary,
+            _gasPrice,
+            _gasLimit,
+            nonce,
+            _hashLock
+        );
+    }
+
+    /**
+     * @notice Transfers EIP20 token to destination address.
+     *
+     * @param _token EIP20 token address.
+     * @param _to Address to which tokens are transferred.
+     * @param _value Amount of tokens to be transferred.
+     */
+    function transferToken(
+        EIP20Interface _token,
```

Comment:
Since token can be any EIP20 token, require can be added that token should not be address 0.
Also please add respective unit case.

Reply:
It is about the `_token` being `address(0).

Reply:
> It is about the _token being address(0). Good question about _to, though 🤔

Yes, we can never be sure if the passed address is of valid EIP20 contract, nor is `address(0)`. 
We had this check in `GatewayComposer`, so we can retain the check for `address(0)`. 

`_to` can be `address(0)`

Reply:
ok I will add the redundant `require` 👍